packages/web-components/fast-ssr/src/element-renderer/element-renderer.ts

```diff
@@ -53,22 +120,25 @@ export class FASTElementRenderer extends ElementRenderer {
      * Render the component internals to shadow DOM.
      * @param renderInfo - information about the current rendering context.
      */
-    *renderShadow(renderInfo: RenderInfo): IterableIterator<string> {
-        // TODO - this will yield out the element's template using the template renderer
-        yield "";
-    }
+    public *renderShadow(renderInfo: RenderInfo): IterableIterator<string> {
+        const view = this.element.$fastController.view;
+        const styles = this.element.$fastController.styles;
 
-    /**
-     * Indicate to the {@link FASTElementRenderer} that an attribute has been changed.
-     * @param name - The name of the changed attribute
-     * @param old - The old attribute value
-     * @param value - The new attribute value
-     */
-    attributeChangedCallback(
-        name: string,
-        old: string | null,
-        value: string | null
-    ): void {
-        this.element.attributeChangedCallback(name, old, value);
+        if (styles) {
+            yield this.styleRenderer.render(styles);
+        }
+
+        if (view !== null) {
+            yield* this.templateRenderer.renderOpCodes(
+                ((view as unknown) as SSRView).codes,
+                renderInfo,
+                this.element,
+                // renderShadow can be called by external code that cant know about execution context,
+                // so that data can't come from a method argument. Provide defaultExecutionContext, but I
+                // am unsure if this behavior will *always* match fast-element's behavior. If it doesn't,
+                // we'll need to implement a executionContext lookup.
+                defaultExecutionContext
```

Comment:
I think you are good here when this comes to custom elements. They all have the default execution context. The place where the context changes is in a `repeat` scenario. The views that the repeat creates may not have the same context.

Reply:
Cool, I'll remove the comment then.